### PR TITLE
Updates to lumberjack package released 4 years ago

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This updates to gopkg.in/natefinch/lumberjack.v2 which was released in
mid 2017. This and other tidies result in less dependency declarations
in downstream projects.

For example:
```diff
--- a/go.mod
+++ b/go.mod
@@ -20,7 +20,6 @@ require (
        github.com/mattn/go-sqlite3 v1.14.6
        github.com/mitchellh/go-homedir v1.1.0
        github.com/mitchellh/mapstructure v1.4.1 // indirect
-       github.com/natefinch/lumberjack v2.0.0+incompatible // indirect
        github.com/neo4j/neo4j-go-driver/v4 v4.2.4
        github.com/pelletier/go-toml v1.8.1 // indirect
        github.com/rogpeppe/go-internal v1.7.0 // indirect
@@ -32,8 +31,6 @@ require (
        github.com/spf13/pflag v1.0.5
        github.com/spf13/viper v1.7.1
        github.com/tetratelabs/log v0.0.0-20210128123858-0fba4e2dac7d
-       go.uber.org/multierr v1.6.0 // indirect
-       go.uber.org/zap v1.16.0 // indirect
        golang.org/x/crypto v0.0.0-20210317152858-513c2a44f670 // indirect
        golang.org/x/lint v0.0.0-20201208152925-83fdc39ff7b5 // indirect
        golang.org/x/mod v0.4.1 // indirect
@@ -46,3 +43,5 @@ require (
        gopkg.in/ini.v1 v1.62.0 // indirect
        gopkg.in/yaml.v2 v2.4.0
 )
+
+replace github.com/tetratelabs/log => ../log

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,21 +1,12 @@
...
-	go.uber.org/zap v1.9.1
-	google.golang.org/grpc v1.15.0
-	gopkg.in/natefinch/lumberjack.v2 v2.0.0 // indirect
-	gopkg.in/yaml.v2 v2.2.1 // indirect
+	github.com/spf13/cobra v1.1.3
+	github.com/spf13/pflag v1.0.5
+	go.uber.org/multierr v1.6.0 // indirect
+	go.uber.org/zap v1.16.0
+	google.golang.org/grpc v1.36.0
+	gopkg.in/natefinch/lumberjack.v2 v2.0.0
```
Comment (codefromthecrypt) [inline go.mod:11]: changing the lumberjack package, then doing `go get -u ./...; go mod tidy` make this file simpler

**Comment by codefromthecrypt** [comment]
ps another (preferred) option could be to remove lumberjack entirely. `istio/mixer/adapter/stdio/zap.go` used to use it, but currently that doesn't seem relevant...

**Review by nacx — APPROVED** [review]
Nice! Mind upgrading the dependency in ngac and the monorepo after this is merged?

**Comment by codefromthecrypt** [comment]
opened #7 about the chopping of lumberjack

**Comment by codefromthecrypt** [comment]
@llinder @johnlanda ps tech debt like this is a good reason to consider whether the functionality in this lib is worthwhile bringing into another project (especially in an OSS one)

**Pull request merged**